src/Catalog/Persistence/Storage.cs

```diff
@@ -170,6 +170,18 @@ public Uri ResolveUri(string relativeUri)
             return new Uri(BaseAddress, relativeUri);
         }
 
+        /// <summary>
+        /// It will return true if there are changes between the source and destination.
+        /// </summary>
+        /// <param name="resourceUri">The uri for the file source.</param>
+        /// <param name="id">The id of the source.</param>
+        /// <param name="version">The version of the source.</param>
+        /// <returns></returns>
+        public virtual Task<bool> IsUpdateRequiredAsync(Uri resourceUri, string id, string version)
```

Comment:
I'm not a fan of this method. It seems very specific to a Catalog2Dnx scenario and I'm not sure where else it would be used.

IMO we should make a method that returns the results of `FetchAttributesAsync` and then have the logic for comparing the hashes inside Catalog2Dnx. If we discover that logic is needed elsewhere, we can create some sort of helper/utility method, but I don't think comparing MD5 hashes belongs in this storage API.

Reply:
Catalog2DNX needs to support multiple Storage types. Currently these are FiltSystem and Azure.
For this reason I think that the logic for deciding if the two resources are out of sync and the destination needs to be updated needs to stay at the Storage level. I updated the method name and signature.